refactor(decorators): change color of 'Working...' error message

MynahMarie opened the pull request:
In this PR I have:

- Modified the baseView in order to filter out the error displaying the text '**Working...**' and give it the class of `info` so that it will display blue instead of red.

- Refactored the notifyDelayedRequest function to reflect the change of class made in the baseView.

fixes #5354

GitCop commented:
There were the following issues with your Pull Request

 * Commit: 909f91790969233a299c5530eb213a457bc8a67f
  * Your subject line is longer than 100 characters

Guidelines are available at https://github.com/mozilla/fxa-content-server/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

MynahMarie commented:
@vladikoff This is more of a quick fix than anything, I can see potentially a few concerns with this but decided to make a PR anyway and maybe get some of your feedback. My main issue is that because of the change of class, this replaces the previous message in that div so I'm not at all sure that this is the behaviour we are aiming for... Let me know what you think and if you know of a better strategy to adopt for this problem.

vladikoff commented:
> because of the change of class

can we add a class if `Working...`, remove the class if any other error?

MynahMarie commented:
@vladikoff yes that was my original idea... working to understand the code base better.

GitCop commented:
There were the following issues with your Pull Request

 * Commit: 909f91790969233a299c5530eb213a457bc8a67f
  * Your subject line is longer than 100 characters

Guidelines are available at https://github.com/mozilla/fxa-content-server/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

MynahMarie commented:
Hi @vladikoff I've added a class for this element and I added the styles for it in main.css but I see that all the .css files are gitignored in this repo... Just for clarity, this is what I added in the main.css:

```
.working-error {
    background: #0a84ff !important; }
```
If you don't mind I have a small question about a problem I came across while experimenting, every time that I restart the server on my machine, the changes I made to the main.css file are reverted... I hope this is not too much of a basic question but do you mind telling me what causes this?

GitCop commented:
There were the following issues with your Pull Request

 * Commit: 909f91790969233a299c5530eb213a457bc8a67f
  * Your subject line is longer than 100 characters

Guidelines are available at https://github.com/mozilla/fxa-content-server/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

GitCop commented:
There were the following issues with your Pull Request

 * Commit: 909f91790969233a299c5530eb213a457bc8a67f
  * Your subject line is longer than 100 characters

Guidelines are available at https://github.com/mozilla/fxa-content-server/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

vladikoff commented on app/scripts/views/base.js:
```diff
@@ -81,11 +81,15 @@ define(function (require, exports, module) {
     this.hideSuccess();
 
     err = this._normalizeError(err);
-
```
Undo this line change

vladikoff commented on app/scripts/views/base.js:
```diff
@@ -81,11 +81,15 @@ define(function (require, exports, module) {
     var $error = this.$('.error');
+
+    if (err.errno === 1005) {
+      $error.addClass('working-error');
```
👍 good workaround

vladikoff commented on app/styles/_state.scss:
```diff
@@ -48,6 +48,10 @@
 }
 
+.working-error {
+  background: $info-background-color !important;
```
@MynahMarie is there any way to make this work without an `!important` ? 

see above on line 41 `&.visible {` can we add a similar block such as `&.working-error {`

MynahMarie replied:
Sure I will give it a try! I used `!important` to override the background property of the `.error` class but I will see if I manage to use `&.visible` instead.

vladikoff commented:
>If you don't mind I have a small question about a problem I came across while experimenting, every time that I restart the server on my machine, the changes I made to the main.css file are reverted... I hope this is not too much of a basic question but do you mind telling me what causes this?

> EDIT: So sorry, that was a dumb question, changed the sass file instead of the main.css. I've never used sass before so that's why I got a bit confused. Now I understand that sass compiles down to css and that's why my changes in the main.css files kept being deleted.

Currently we open up a new terminal tab and use `grunt sass watch` to watch for changes to CSS

vladikoff commented:
Feel free to reopen this with a fix to https://github.com/mozilla/fxa-content-server/pull/5983#discussion_r175836034 if you have time
